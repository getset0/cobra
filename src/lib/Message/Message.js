import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { theme } from "../constants";
import Card from "../Card/Card";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
  faExclamationCircle,
  faTimesCircle,
  faInfoCircle,
  faCheckCircle
} from "@fortawesome/free-solid-svg-icons";

const iconStyles = {
  fontSize: "28px",
  marginRight: "10px"
};

const MessageInnerWrapper = styled.div`
  display: flex;
  align-items: center;
`;

const Message = ({ type, text, hasShadow, ...props }) => {
  const messageTypes = {
    warning: (
      <Card
        backgroundColor="khaki"
        hasShadow={hasShadow}
        textColor="tomato"
        {...props}
      >
        <MessageInnerWrapper>
          <FontAwesomeIcon style={iconStyles} icon={faExclamationCircle} />
          {text}
        </MessageInnerWrapper>
      </Card>
    ),
    error: (
      <Card
        backgroundColor="lightcoral"
        hasShadow={hasShadow}
        textColor="firebrick"
        {...props}
      >
        <MessageInnerWrapper>
          <FontAwesomeIcon style={iconStyles} icon={faTimesCircle} />
          {text}
        </MessageInnerWrapper>
      </Card>
    ),
    success: (
      <Card
        backgroundColor="lightgreen"
        hasShadow={hasShadow}
        textColor="seagreen"
        {...props}
      >
        <MessageInnerWrapper>
          <FontAwesomeIcon style={iconStyles} icon={faCheckCircle} />
          {text}
        </MessageInnerWrapper>
      </Card>
    ),
    default: (
      <Card
        backgroundColor="lightblue"
        hasShadow={hasShadow}
        textColor="royalblue"
        {...props}
      >
        <MessageInnerWrapper>
          <FontAwesomeIcon style={iconStyles} icon={faInfoCircle} />
          {text}
        </MessageInnerWrapper>
      </Card>
    )
  };
  return messageTypes[type];
};

Message.propTypes = {
  type: PropTypes.oneOf([ "warning", "error", "success", "default" ]),
  width: PropTypes.number,
  theme: PropTypes.object,
  text: PropTypes.string,
  hasShadow: PropTypes.bool
};

Message.defaultProps = {
  theme: theme,
  hasShadow: true,
  type: "default"
};

export default Message;
