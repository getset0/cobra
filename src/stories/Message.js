import React from "react";
import { storiesOf } from "@storybook/react";
import { withInfo } from "@storybook/addon-info";
import { withBackgrounds } from "@storybook/addon-backgrounds";
import { backgroundOptions } from "./util";
import { Message } from "../lib";

storiesOf("Message", module)
  .addDecorator(withInfo)
  .addDecorator(withBackgrounds(backgroundOptions))
  .add("Messages", () => (
    <React.Fragment>
      <Message text="I am a default" />
      <Message text="I am a warning" type="warning" />
      <Message text="I am a success" type="success" />
      <Message text="I am an error" type="error" />
      <Message text="Test" width={100} />

      <Message text="I am a default" hasShadow={false} />
      <Message text="I am a warning" type="warning" hasShadow={false} />
      <Message text="I am a success" type="success" hasShadow={false} />
      <Message text="I am an error" type="error" hasShadow={false} />
      <Message text="Test" width={100} hasShadow={false} />
    </React.Fragment>
  ));
