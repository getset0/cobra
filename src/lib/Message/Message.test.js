import React from "react";
import Message from "./Message";
import "jest-styled-components";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

it("renders message without crashing", () => {
  shallow(<Message />);
});

it("renders default message correctly", () => {
  const wrapper = shallow(<Message />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});

it("renders warning message correctly", () => {
  const wrapper = shallow(<Message type="warning" />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});

it("renders success message correctly", () => {
  const wrapper = shallow(<Message type="success" />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});

it("renders error message correctly", () => {
  const wrapper = shallow(<Message type="error" />);
  const tree = toJson(wrapper);
  expect(tree).toMatchSnapshot();
});
