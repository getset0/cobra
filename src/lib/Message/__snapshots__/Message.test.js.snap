// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders default message correctly 1`] = `
<Styled(CustomCard)
  backgroundColor="lightblue"
  textColor="royalblue"
  theme={
    Object {
      "blackColor": "#000000",
      "darkBlueColor": "#02469B",
      "darkGrayColor": "#4A4A4A",
      "darkestBlueColor": "#02254D",
      "darkestGrayColor": "#303030",
      "greenColor": "#76B041",
      "lightBlueColor": "#0190D0",
      "lightGrayColor": "#D8D8D8",
      "lightestBlueColor": "#00FEFE",
      "mediumGrayColor": "#979797",
      "primaryColor": "#1676B7",
      "secondaryColor": "#17BEBB",
      "whiteBackgroundColor": "#fdfdfd",
      "whiteColor": "#ffffff",
      "yellowColor": "#FFC914",
    }
  }
>
  <styled.div>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f05a",
            "M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z",
          ],
          "iconName": "info-circle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "fontSize": "28px",
          "marginRight": "10px",
        }
      }
      symbol={false}
      title=""
      transform={null}
    />
  </styled.div>
</Styled(CustomCard)>
`;

exports[`renders error message correctly 1`] = `
<Styled(CustomCard)
  backgroundColor="lightcoral"
  textColor="firebrick"
  theme={
    Object {
      "blackColor": "#000000",
      "darkBlueColor": "#02469B",
      "darkGrayColor": "#4A4A4A",
      "darkestBlueColor": "#02254D",
      "darkestGrayColor": "#303030",
      "greenColor": "#76B041",
      "lightBlueColor": "#0190D0",
      "lightGrayColor": "#D8D8D8",
      "lightestBlueColor": "#00FEFE",
      "mediumGrayColor": "#979797",
      "primaryColor": "#1676B7",
      "secondaryColor": "#17BEBB",
      "whiteBackgroundColor": "#fdfdfd",
      "whiteColor": "#ffffff",
      "yellowColor": "#FFC914",
    }
  }
>
  <styled.div>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f057",
            "M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z",
          ],
          "iconName": "times-circle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "fontSize": "28px",
          "marginRight": "10px",
        }
      }
      symbol={false}
      title=""
      transform={null}
    />
  </styled.div>
</Styled(CustomCard)>
`;

exports[`renders success message correctly 1`] = `
<Styled(CustomCard)
  backgroundColor="lightgreen"
  textColor="seagreen"
  theme={
    Object {
      "blackColor": "#000000",
      "darkBlueColor": "#02469B",
      "darkGrayColor": "#4A4A4A",
      "darkestBlueColor": "#02254D",
      "darkestGrayColor": "#303030",
      "greenColor": "#76B041",
      "lightBlueColor": "#0190D0",
      "lightGrayColor": "#D8D8D8",
      "lightestBlueColor": "#00FEFE",
      "mediumGrayColor": "#979797",
      "primaryColor": "#1676B7",
      "secondaryColor": "#17BEBB",
      "whiteBackgroundColor": "#fdfdfd",
      "whiteColor": "#ffffff",
      "yellowColor": "#FFC914",
    }
  }
>
  <styled.div>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f058",
            "M504 256c0 136.967-111.033 248-248 248S8 392.967 8 256 119.033 8 256 8s248 111.033 248 248zM227.314 387.314l184-184c6.248-6.248 6.248-16.379 0-22.627l-22.627-22.627c-6.248-6.249-16.379-6.249-22.628 0L216 308.118l-70.059-70.059c-6.248-6.248-16.379-6.248-22.628 0l-22.627 22.627c-6.248 6.248-6.248 16.379 0 22.627l104 104c6.249 6.249 16.379 6.249 22.628.001z",
          ],
          "iconName": "check-circle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "fontSize": "28px",
          "marginRight": "10px",
        }
      }
      symbol={false}
      title=""
      transform={null}
    />
  </styled.div>
</Styled(CustomCard)>
`;

exports[`renders warning message correctly 1`] = `
<Styled(CustomCard)
  backgroundColor="khaki"
  textColor="tomato"
  theme={
    Object {
      "blackColor": "#000000",
      "darkBlueColor": "#02469B",
      "darkGrayColor": "#4A4A4A",
      "darkestBlueColor": "#02254D",
      "darkestGrayColor": "#303030",
      "greenColor": "#76B041",
      "lightBlueColor": "#0190D0",
      "lightGrayColor": "#D8D8D8",
      "lightestBlueColor": "#00FEFE",
      "mediumGrayColor": "#979797",
      "primaryColor": "#1676B7",
      "secondaryColor": "#17BEBB",
      "whiteBackgroundColor": "#fdfdfd",
      "whiteColor": "#ffffff",
      "yellowColor": "#FFC914",
    }
  }
>
  <styled.div>
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon={
        Object {
          "icon": Array [
            512,
            512,
            Array [],
            "f06a",
            "M504 256c0 136.997-111.043 248-248 248S8 392.997 8 256C8 119.083 119.043 8 256 8s248 111.083 248 248zm-248 50c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z",
          ],
          "iconName": "exclamation-circle",
          "prefix": "fas",
        }
      }
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      style={
        Object {
          "fontSize": "28px",
          "marginRight": "10px",
        }
      }
      symbol={false}
      title=""
      transform={null}
    />
  </styled.div>
</Styled(CustomCard)>
`;
